src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs

```diff
@@ -729,7 +729,8 @@ private async Task<Entity> GetEntity(GlobalAddress address, GatewayModel.LedgerS
             return entity;
         }
 
-        if (address.ToString()[0] == _ecdsaSecp256k1VirtualAccountAddressPrefix || address.ToString()[0] == _eddsaEd25519VirtualAccountAddressPrefix)
+        var decodedAddress = RadixAddressCodec.Decode(address);
+        if (decodedAddress.Data[0] == _ecdsaSecp256k1VirtualAccountAddressPrefix || decodedAddress.Data[0] == _eddsaEd25519VirtualAccountAddressPrefix)
         {
             return new VirtualAccountComponentEntity(address);
         }
```

Comment:
BTW when did we lost support for VirtualIdentities?!

Reply:
well, that would be when I introduced that global address type.

Reply:
fixed.